src/main/java/gregtech/api/capability/impl/FuelRecipeLogic.java

```diff
@@ -207,21 +224,88 @@ public long getMaxVoltage() {
         return maxVoltage;
     }
 
+    protected double calculateFuelConsumptionMultiplier() {
+        return 1.0;
+    }
+
     protected int calculateFuelAmount(FuelRecipe currentRecipe) {
-        return currentRecipe.getRecipeFluid().amount * getVoltageMultiplier(getMaxVoltage(), currentRecipe.getMinVoltage());
+        return (int) (currentRecipe.getRecipeFluid().amount * getVoltageMultiplier(getMaxVoltage(), currentRecipe.getMinVoltage()) * calculateFuelConsumptionMultiplier());
+    }
+
+    public int getFuelAmount() {
+        return calculateFuelAmount(this.previousRecipe);
+    }
+
+    protected double calculateRecipeDurationMultiplier() {
+        return 1.0;
     }
 
     protected int calculateRecipeDuration(FuelRecipe currentRecipe) {
-        return currentRecipe.getDuration();
+        return (int) (currentRecipe.getDuration() * calculateRecipeDurationMultiplier());
+    }
+
+    public int getRecipeDuration() {
+        return calculateRecipeDuration(this.previousRecipe);
+    }
+
+    /**
+     * this method aim to tell if the current recipe has ended in order to check for a new recipe see {@link #isActive}
+     * or {@link #isWorkingEnabled} if you want to know if the machine is running.
+     *
+     * @return true if no recipe is running false otherwise
+     */
+    public boolean hasRecipeEnded() {
```

Comment:
This method is pointless and badly designed. Why would you EVER want to override it? 
If you want it to prevent machine from picking up new recipe, you have isReadyForRecipes
If you want to stop machine from producing energy, you have canProduceEnergy
If you want to affect recipe duration, you have calculateRecipeDuration

Reply:
>This method is pointless and badly designed. Why would you EVER want to override it? 

>Additional info:
see #1604 for additional information on why and for what those changes are made

Reply:
I don't see this method overriden in large turbine in this PR. So why is it needed?

Reply:
>I don't see this method overriden in large turbine in referenced PR. So why is it needed?

read the comments not the code this methods doesn't yet exist of course it can't be overriden. this part has been extensively discussed

Reply:
Still makes no sense to introduce method for something so niche and resolvable by other means

Reply:
>Still makes no sense to introduce method for something so niche and resolvable by other means

find a way to resolve the problem it was intended to fix and post it as a comment in #1604 if you are able to do so then this method will become useless I'll be glad to not to have to change RecipeDurationLeft to a float

Reply:
Alright, judging from the code in your turbine PR, you want to compare it against getRecipeDurationMultiplier() and not zero. But the question is, why can't you just let it tick for one more time and decrease remaining ticks by getRecipeDurationMultiplier(), making it zero and not needing any changes?
I really don't see why it should REALLY stop when value is less than multiplier and cannot work for one more tick and then stop when it's actually zero or below.

Reply:
the previous code was using `== 0` not `<= 0` so yeah. 
This method would still be useful to stop the current recipe on other conditions than just recipe duration.

>I really don't see why it should REALLY stop when value is less than multiplier and cannot work for one more tick and then stop when it's actually zero or below.

just to not overproducing by one tick yes this is little but I don't want the new implementation to contain known and ignored bugs.